com.unity.multiplayer.mlapi/Tests/Editor/SnapshotRttTests.cs

```diff
@@ -0,0 +1,75 @@
+using MLAPI;
+using NUnit.Framework;
+using UnityEngine;
+
+namespace MLAPI.EditorTests
+{
+    public class SnapshotRttTests
+    {
+        private const double k_Epsilon = 0.0001;
+
+        [Test]
+        public void TestBasicRtt()
+        {
+            var snapshot = new SnapshotSystem();
+            var client1 = snapshot.Rtt(0);
+
+            client1.NotifySend(0, 0.0);
+            client1.NotifySend(1, 10.0);
+
+            client1.NotifyAck(1, 15.0);
+
+            client1.NotifySend(2, 20.0);
+            client1.NotifySend(3, 30.0);
+            client1.NotifySend(4, 32.0);
+
+            client1.NotifyAck(4, 38.0);
+            client1.NotifyAck(3, 40.0);
+
+            ConnectionRtt.Rtt ret = client1.GetRtt();
+            Assert.True(ret.AverageSec < 7.0 + k_Epsilon);
+            Assert.True(ret.AverageSec > 7.0 - k_Epsilon);
+            Assert.True(ret.WorstSec < 10.0 + k_Epsilon);
+            Assert.True(ret.WorstSec > 10.0 - k_Epsilon);
+            Assert.True(ret.BestSec < 5.0 + k_Epsilon);
+            Assert.True(ret.BestSec > 5.0 - k_Epsilon);
+
+            // note: `last` latency is latest received Ack, not latest sent sequence.
+            Assert.True(ret.LastSec < 10.0 + k_Epsilon);
+            Assert.True(ret.LastSec > 10.0 - k_Epsilon);
+        }
+
+        [Test]
+        public void TestEdgeCasesRtt()
+        {
+            var snapshot = new SnapshotSystem();
+            var client1 = snapshot.Rtt(0);
+            var iterationCount = 200;
```

Comment:
Should this be in terms of `k_RingSize` perhaps?  (like `k_RingSize * 3` ?)

Reply:
```suggestion
            var iterationCount = k_RingSize * 3;
```

Reply:
`k_RingSize` is private, and afaik, c# doesn't have `friend`. This raise a few possibilities:
- make the ring size a publicly available setting with a setter, and getter
- only expose the ring size with a readonly getter
- leave the test with hard-coded values, as-is.

Do you have a preference?

Reply:
we have `internal` in C# and internals are exposed to test asmdefs IIRC